exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java

```diff
@@ -106,19 +177,30 @@ public HoldingContainer visitFunctionCall(FunctionCall call, ClassGenerator<?> g
     @Override
     public HoldingContainer visitBooleanOperator(BooleanOperator op,
         ClassGenerator<?> generator) throws RuntimeException {
+      HoldingContainer hc = getPrevious(op, generator.getMappingSet());
```

Comment:
You put the check whether it's common subexpression in EvalVisitor, when handling with different expression. 

I feel the code would be much cleaner, if we add a CSEFilter in a similar to ConstantFilter.  The logic will go through ConstantFilter -> CSEFilter -> EvalVisitor.  The check of common subexpression and adding to the Hashmap would be done solely in CSEFilter:visitiUnknow().  In AbstractExrVisitor,  other methods will by default come to visitUnknow().  In this way, there is no need to repeat the checking of common subexpression for different expression in EvalVisitor; actually no change has to be done to EvalVisitor.